demoapp/demo_app_functions.py

```diff
@@ -450,4 +466,479 @@ def update_csv(bill_num: str, title: str, summarized_bill: str, category: str, t
+
+                Response Format:\n
+                Summary: [SUMMARY]\n
+
+                The bill title is: \"{title}"\
+
+                The bill text is: \"{context}"\
+
+                The relevant section names are: \"{names}"\
+
+                The relevant section text is: \"{mgl_sections}"\
+                
+                The relevant committee information: \"{committee_info}"\
+                """
+        bill_details.invoke_dict = prepare_invoke_dict(bill_details)
+
+    return query
+
+def get_query_for_categorizing(bill_details: BillDetails, llm_call_type: str) -> str: 
+
+    """
+        
+    This functions prepares a prompt based on the call type (small: No use of RAG, large: Use RAG) for bill categoruzation
+
+    Args: 
+        bill_details (BillDetails): Object containing information about the bill - bill_text, bill_title, mgl_ref, commottee_info, mgl_names, category_list(If new categories things are added)
+        llm_call_type (str): This argument can take 2 values ("small": No use of RAG, "large": Use RAG) 
+
+    Returns: 
+        str: Query that includes task, context and instructions on bill categorization
+
+    """
+
+    if llm_call_type == 'large':
+        query = f""" Your Job here is to identify what categories the following MA Legislative bill belongs to. 
+
+                    Note that the bill refers to specific existing chapters and sections of the Mass General Laws. Use the information from those chapters and sections in your context to categorize\n
+                    Categorize the bill that reads as follows:\n{getattr(bill_details, 'bill_text')}\n and is titled: {getattr(bill_details, 'bill_title')}\n
+                    
+                    Classify the bill according to this list of {getattr(bill_details, 'categories', new_categories_for_bill_list)} to the closest relevant categories. Do not output categories outside of the list. \n
+                    No need for any explanation on why you choose those specific categories. 
+                    
+                    Use the information from those chapters and sections in your context as well as the corresponding {getattr(bill_details, "mgl_names")} of those chapter and sections to generate categories.\n 
+                    
+                    Also use committee information {getattr(bill_details, 'committee_info')} if available.\n
+
+                    INSTRUCTIONS: 
+                    1. Choose no more than 2 categories from the list above.
+                    2. Do not provide explanations for the category choices.
+                    3. Do not output categories not listed above.
+                    4. Do not modify or paraphrase the category names, choose directly from the list provided.
+
+                    Response Format:
+                
+                    Categories:CATEGORIES
```

Comment:
I wonder if we need to be more specific about the category format to get well formatted responses. Again LangChain function calling utilities can help here.